src/shared/XLinkDispatcherImpl.c

```diff
@@ -476,20 +476,27 @@ void dispatcherCloseLink(void* fd, int fullClose)
 
     for (int index = 0; index < XLINK_MAX_STREAMS; index++) {
         streamDesc_t* stream = &link->availableStreams[index];
-        if (!stream) {
+        // Wasn't initialized
+        if(stream->id == INVALID_STREAM_ID) {
             continue;
         }
 
         // TODO integrate pending changes
         // * use move semantic, this prevents the memset(0) from causing leak/crash
         // * make new xlink-specific semaphore and wait on it during xlink lookup, create, etc.
+
         XLink_sem_wait(&stream->sem);
```

Comment:
This needs to be something like `if (XLink_sem_wait(...) < 0) continue;`
It is possible another thread has destroyed the semaphore. In that scenario, XLink_sem_wait() will return -1. Therefore, this iteration in the loop should be skipped.

Reply:
fyi, I needed more time to reply, because I'm separating this PR converation from the issue of incorrect/bug XLink_Semaphore. TL;DR...xlink is improperly using a semaphore with streams when it should be using a mutex. A separate issue on this is pending.